pkg/service/credential/service.go

```diff
@@ -90,9 +88,44 @@ func NewCredentialService(config config.CredentialServiceConfig, s storage.Servi
 }
 
 func (s Service) CreateCredential(ctx context.Context, request CreateCredentialRequest) (*CreateCredentialResponse, error) {
-	returnFunc := s.createCredentialFunc(request)
+	watchKeys := make([]storage.WatchKey, 0)
+	var statusListCredential *StoredCredential
+
+	var slcMetadata StatusListCredentialMetadata
+
+	if request.hasStatus() && request.isStatusValid() {
+
+		var statusListUUID string
+		var err error
+
+		statusPurpose := statussdk.StatusRevocation
+
+		if request.Suspendable {
+			statusPurpose = statussdk.StatusSuspension
+		}
+
+		statusListUUID, statusListCredential, err = s.storage.GetStatusListCredentialKeyData(ctx, request.Issuer, request.JSONSchema, statusPurpose)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting status list watch key uuid data")
+		}
+
+		if statusListUUID == "" {
+			return nil, errors.New("generating uuid")
+		}
+
+		statusListCredentialWatchKey := s.storage.GetStatusListCredentialWatchKey(request.Issuer, request.JSONSchema, string(statusPurpose))
+		statusListCredentialIndexPoolWatchKey := s.storage.GetStatusListIndexPoolWatchKey(request.Issuer, request.JSONSchema, string(statusPurpose))
+		statusListCredentialCurrentIndexWatchKey := s.storage.GetStatusListCurrentIndexWatchKey(request.Issuer, request.JSONSchema, string(statusPurpose))
+
+		watchKeys = append(watchKeys, statusListCredentialWatchKey)
+		watchKeys = append(watchKeys, statusListCredentialIndexPoolWatchKey)
```

Comment:
Would you mind helping me understand if can be removed `statusListCredentialIndexPoolWatchKey` from the watch keys? 

My reading of the code is that the status-list-index-pool is written once, and then never updated. It is written when the status list is created for the first time. If it it never modified, then it will not affect the execution of a transaction. Does that make sense?

Reply:
Well the first time it is created it needs a watch key.

thread one
status-list-index-pool:issuer1,schema1,purpose1

thread two
status-list-index-pool:issuer1,schema1,purpose1

thead one creates the pool and assigns it to a cred,

thread two is about to create the cred, but it is watched so it will optimistically fail.

Something is not working properly in our watch key stuff as per this issue, so I will be revisiting the entire mechanism...
https://github.com/TBD54566975/ssi-service/issues/302